### ISSUE TITLE
Port in deployment server resolves to `0` in Docker driver [docker driver]

### ISSUE THREAD

**Issue by tnolet** [magneticio/vamp#121]
Using the following blueprint:

```yaml
name: monarch:1.0

clusters:

  monarchs:
    services:
      breed:
        name: monarch:0.2
        deployable: magneticio/monarch:0.2
        ports:
          port: 8080/http
    
      scale:
          cpu: 1
          memory: 1024
          instances: 1
```

The result is a server which maps port `8080` to `0`. The deployment looks as follows:

```json
[
    {
        "name": "06b6ac51-100c-4d00-842a-8458cea0142c",
        "endpoints": {},
        "clusters": {
            "monarchs": {
                "services": [
                    {
                        "state": {
                            "name": "Deployed",
                            "started_at": "2015-05-24T16:04:33.032Z"
                        },
                        "breed": {
                            "name": "monarch:0.2",
                            "deployable": "magneticio/monarch:0.2",
                            "ports": {
                                "port": "8080/http"
                            },
                            "environment_variables": {},
                            "constants": {},
                            "dependencies": {}
                        },
                        "scale": {
                            "cpu": 1,
                            "memory": 1024,
                            "instances": 1
                        },
                        "routing": {
                            "weight": 100,
                            "filters": []
                        },
                        "servers": [
                            {
                                "name": "b45a0c1f690fa750179e",
                                "host": "localhost",
                                "ports": {
                                    "8080": 0
                                },
                                "deployed": true
                            }
                        ],
                        "dependencies": {},
                        "dialects": {}
                    }
                ],
                "routes": {
                    "8080": 33005
                },
                "dialects": {}
            }
        },
        "ports": {
            "monarchs.port": "33005"
        },
        "environment_variables": {},
        "constants": {},
        "hosts": {
            "monarchs": "localhost"
        }
    }
]
```

In router, it looks as follows:

```json
[
    {
        "name": "06b6ac51-100c-4d00-842a-8458cea0142c_monarchs_8080",
        "port": 33005,
        "protocol": "http",
        "httpQuota": {},
        "tcpQuota": {},
        "filters": [],
        "services": [
            {
                "name": "monarch:0.2",
                "weight": 100,
                "servers": [
                    {
                        "name": "b45a0c1f690fa750179e",
                        "host": "localhost",
                        "port": 0
                    }
                ]
            }
        ]
    }
]
```

**Comment by plamola** [comment]
Issue only applies when the docker image has no exposed ports.

**Comment by plamola** [comment]
Possible solution would be to add the ports specified in the breed to the exposed ports of the container. That way, docker would be able to assign a random host port to it.

Current tugboat implementation does not map the exposed ports to the request. This needs to be fixed first, see #119

**Comment by tnolet** [comment]
@plamola lets discuss thiis for 0.7.8
